Images/ImageImportExport/ImageExport.cpp

```diff
@@ -0,0 +1,249 @@
+	// The image's number of color components as been added to PDEImageAttrs.
+	ASInt32 numComps = attrs.numComponents;
+
+	// There is also a DLPDEImageSetIntent() function.
+	const char* renderIntent = DLPDEImageGetIntent(&image);
+	
+	sprintf(message, "PDEImage data:\n\tWidth:\t%f\n\tHeight:\t%f\n\tNumber of components:\t%i\n"
+	"\tBits per component\t\%i\n\tRender Intent:\t%s\n", width, height, numComps, bpc, renderIntent);
+	std::cout << message << std::endl;
+}
+
+char* CompressionToString(DLImageCompression compression){
+	char * comp = "";
+	switch (compression) {
+    case NONE:
+        comp = "NONE";
+        break;
+    case FLATE:
+        comp = "FLATE";
+        break;
+    case LZW:
+        comp = "LZW";
+        break;
+    case G3:
+        comp = "G3";
+        break;
+    case G4:
+        comp = "G4";
+        break;
+    case DCT:
+        comp = "DCT";
+        break;
+    default:
+        comp = "Default";
+        break;
+    }
+	
+	return comp;
+}
+
+// This function provides detailed information about the PDEImage's DLPDEImageExportParams.
+void PrintExportInfo(const PDEImage image, const DLPDEImageExportParams &params) {
+	// Get the compression type used to import the PDEImage. This is not part of DLPDEImageExportParams, but can be nice to know.
+    DLImageCompression importCompression = DLPDEImageGetCompression(image);
+
+	// Get the compression type that will be used for exporting the PDEImage.
+	DLImageCompression exportCompression = params.TIFFCompression;
+
+    char *impComp = CompressionToString(importCompression);
+	char *expTIFComp = CompressionToString(exportCompression);
+	
+	char message[1024];
+    sprintf(message,
+            "DLPDEImageExportParams:\n\tReverseGray = %s,\n\tTIFFBlackIsOne = %s,\n\tHalftoneGrayImages = %s,\n"
+            "\tJPEGQuality = %d,\n\tExportDPI = %d,\n\tTIFFCompression = %s,\n\tImportedCompressionType = %s,\n\tMonochromeThreshold = %s,\n"
+            "\tTIFFCMYKAlpha = %s,\n\tRaiseExceptionIfNotUsableColor = %s,\n"
+            "\tWriteSeparationColorsToTIFF = %s,\n\tDisableAlphaChannelGeneration = %s\n",
+            BTS(params.ReverseGray), BTS(params.TIFFBlackIsOne), BTS(params.HalftoneGrayImages),
+            params.JPEGQuality, params.ExportDPI, expTIFComp, impComp, BTS(params.MonochromeThreshold), BTS(params.TIFFCMYKAlpha),
+            BTS(params.RaiseExceptionIfNotUsableColor), BTS(params.WriteSeparationColorsToTIFF),
+            BTS(params.DisableAlphaChannelGeneration));
+    
+	std::cout << message << std::endl;
+}
+
+// This function loops throught all PDEContent objects inside a given page and searches for PDEImage elements.
+void FindAndExportImages(PDEContent content, int pageNum)
+{
+    int numElements = PDEContentGetNumElems(content);
+    int element = 0;
+	
+    // Loop throught every element on the page.
+    while (element < numElements) 
+   {
+        PDEElement contentElement = PDEContentGetElem(content, element);
+        
+        /*
```

Comment:
You need to adjust the alignment of the comment to match the opening and closing.